### PR TITLE
Mostly kill off openshift-sdn's node-config.yaml

### PR DESCRIPTION
**Pull request by danwinship**
Update openshift-sdn to get all of the "sdn"/"node" configuration from the ClusterNetwork object, and to optionally get the kube-proxy configuration from a kube-proxy config yaml file instead of the old node-config.yaml. This sets us up to remove the old node-config.yaml entirely.

Without node-config.yaml there's no way to configure whether unidling should be enabled, but this isn't actually a regression since we didn't expose that as an option in CNO anyway. (It just always left the option unset in the node-config, meaning it was always enabled.) There is already a jira card about this (https://jira.coreos.com/browse/SDN-436).

### CODE REVIEW

**Comment by squeed** [comment]
I have most of a PR to expose disabling unidling - so it would be good to have that in this PR. Can you make it a cli flag to openshift-sdn?

File: staging/src/github.com/openshift/sdn/pkg/openshift-sdn/cmd.go [hunk 1]
```diff
@@ -123,47 +128,38 @@ func (sdn *OpenShiftSDN) Run(c *cobra.Command, errout io.Writer, stopCh chan str
...
-
-	sdn.ProxyConfig, err = ProxyConfigFromNodeConfig(*sdn.NodeConfig)
-	if err != nil {
-		klog.V(4).Infof("Unable to build proxy config: %v", err)
-		return err
+		sdn.ProxyConfig.HostnameOverride = sdn.nodeName
+		// FIXME: make this configurable
+		sdn.enableUnidling = true
+	} else {
+		klog.V(2).Infof("Reading proxy configuration from %s", sdn.ConfigFilePath)
```
Comment (squeed) [inline staging/src/github.com/openshift/sdn/pkg/openshift-sdn/cmd.go]: nit: "reading node configuration", but honestly this will be gone in a few days.

**Comment by squeed** [comment]
Besides adding an unidling flag, this is awesome.

**Comment by danwinship** [comment]
> Can you make it a cli flag to openshift-sdn?

We're generally trying to get rid of those aren't we?

It seems like it should go in the proxy config. Actually, it seems like it should specifically go in the "mode" field of the proxy config...

> nit: "reading node configuration", but honestly this will be gone in a few days.

it's reading node-config.yaml, but it's only reading it to get the proxy configuration

**Comment by squeed** [comment]
> It seems like it should go in the proxy config. Actually, it seems like it should specifically go in the "mode" field of the proxy config...

Sure, that's fine (though we could theoretically enable unidling + ipvs, if we get off our butts). Just add the flag in some reasonable way.

**Comment by danwinship** [comment]
OK, unidling is now configured by setting `mode: unidling+iptables`. FTR, it turns out that IPVS wasn't even just "unsupported". It actually didn't work, because our init code only dealt with `iptables` and `userspace`.

**Comment by squeed** [comment]
Looks good to me, but e2e doesn't think so:

```
F0617 16:22:49.202330    2522 cmd.go:115] Failed to initialize sdn: failed to initialize SDN: "redhat/openshift-ovs-networkpolicy" plugin is not compatible with proxy-mode "unidling+iptables"
```

**Comment by danwinship** [comment]
I mean, technically it's right...

**Pull request merged**